### ISSUE TITLE
Wayland crash when creating a new workspace and swiping between workspaces quickly in multitasking view

### ISSUE THREAD

**Issue by wpkelso** [elementary/gala#2200]
### What Happened?

The session appears to crash and restart when a new workspace is created simultaneously with a swipe between workspaces in the multitasking view.

### Steps to Reproduce

1. Open an application
2. Open an application in a new workspace
3. Move one of those applications to a new workspace (drag the application to the new workspace button)
4. Quickly swipe to a workspace

### Expected Behavior

Expected to create the workspace with the application I specified, then move my view over to the workspace I swiped to.

### OS Version

8.x (Circe)

### Session Type

Secure Session (Wayland)

### Software Version

Latest release (I have run all updates)

### Log Output

```shell
(Currently working on getting these)
```

### Hardware Info

![image](https://github.com/user-attachments/assets/4a70e127-993a-463c-afc7-b30a7b595851)

**Comment by leolost2605** [comment]
I got crashes in the multitasking view quite regular (even since the beginning of the OS8 dev cycle) but they were all fixed by #2160 (to be released) so I'm hoping this one might be as well

**Comment by leolost2605** [comment]
In your reproduction steps did one of the workspaces (maybe even the active one) end up empty by chance and you swipe away from it?

**Comment by wpkelso** [comment]
Yes, I was always moving the application in the active workspace to a new one, then switching to the other inactive workspace (not the new space)

**Comment by leolost2605** [comment]
Sounds exactly like my crashes so I'm pretty sure this will be fixed by the next release :)

**Comment by vvvvvvitor** [comment]
Maybe someone can volunteer to test that commit and see if it also fixes this issue?

**Comment by leolost2605** [comment]
I've tried the reproduction steps and can confirm it doesn't crash. I personally usually prefer leaving these prominent issues open until the fix is released to prevent duplicates but we can close it as well :shrug:
